core/runtime/impl/block_builder_impl.cpp

```diff
@@ -19,98 +19,40 @@ namespace kagome::runtime {
   using primitives::CheckInherentsResult;
   using primitives::Extrinsic;
   using primitives::InherentData;
-  using wasm::Literal;
 
   BlockBuilderImpl::BlockBuilderImpl(Buffer state_code,
-                                     std::shared_ptr<Extension> extension)
-      : memory_{extension->memory()},
-        executor_{std::move(extension)},
-        state_code_{std::move(state_code)} {}
+                                     std::shared_ptr<Extension> extension) {
+    runtime_ = std::make_unique<RuntimeApi>(std::move(state_code),
+                                            std::move(extension));
+  }
+
+  BlockBuilderImpl::~BlockBuilderImpl() {}
```

Comment:
```suggestion
  BlockBuilderImpl::~BlockBuilderImpl() = default;
```

Reply:
If I do this, compiler cannot make destructor since it doesn't know the type of RuntimeApi.
The other possible option is to make it base class for apis, but in this case we cannot hide implementation. Maybe it isn't so bad...

Reply:
made all apis inherit RuntimeApi, so we don't need to define destructors in cpp.